GitHubAction/GitHubAction/Factories/CatalogDataFactory.cs

```diff
@@ -0,0 +1,74 @@
+﻿namespace GitHubAction.Factories
+{
+    using Catalog.Domain;
+
+    using GIT;
+
+    using global::GitHubAction.Domain.Entities;
+
+    using Package.Domain.Models;
+
+    using System.Text.RegularExpressions;
+
+    internal class CatalogDataFactory
+    {
+        public static CatalogData Create(Inputs inputs, CreatedPackage createdPackage, IGITInfo git, string sourceUri, string branch, string releaseUri)
+        {
+            if (string.IsNullOrWhiteSpace(branch) || branch == inputs.Version)
+            {
+                branch = git.GetCurrentBranch(inputs.Version);
+            }
+
+            string committerMail = git.GetCommitterMail();
+
+            CatalogData catalog = new CatalogData()
+            {
+                ContentType = createdPackage.Type,
+                Branch = branch,
+                Version = string.IsNullOrWhiteSpace(inputs.Version) ? "0.0.0" : inputs.Version,
+                Identifier = sourceUri,
+                Name = inputs.PackageName ?? "PackageName",
+                CommitterMail = committerMail,
+                ReleaseUri = releaseUri
+            };
+
+            bool isPreRelease = CleanAndAddVersion(inputs, catalog);
+            catalog.IsPreRelease = isPreRelease;
+
+            return catalog;
+        }
+
+        private static bool CleanAndAddVersion(Inputs inputs, CatalogData catalog)
+        {
+            bool isPreRelease;
+            if (!string.IsNullOrWhiteSpace(inputs.BuildNumber))
```

Comment:
build number will only be filled in for prereleases?

Reply:
Build number is an optional argument in the github action. It's filled in when there is no version tag but only a branch and a build number.